Core/Src/main.c

```diff
@@ -87,17 +97,26 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
-  /* USER CODE BEGIN 2 */
+  MX_SPI1_Init();
+  MX_TIM1_Init();
 
+  /* USER CODE BEGIN 2 */
+  HAL_TIM_PWM_Init(&htim1); //start PWM
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_SPI_TransmitRecieve_IT(&hspi1, &spi_tx_buf, &spi_rx_buf, sizeof(spi_tx_buf));
+	  adc_val = ((uint16_t)(spi_recieve_buf[1])) << 8 | ((uint16_t) spi_recieve_buf[2]); //keep 10 MSB
```

Comment:
This is great for getting the 10 bits you need, but it also still has the other 6 bits. Up to you how to remove them, I used bit masking after I had all 16 bits to get just those 10.